src/Resources/RelationManager/AttachRecord.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Filament\Resources\RelationManager;
+
+use Filament\Forms\Components\Select;
+use Filament\Forms\Form;
+use Filament\Forms\HasForm;
+use Illuminate\Support\Str;
+use Livewire\Component;
+
+class AttachRecord extends Component
+{
+    use HasForm;
+
+    public $cancelButtonLabel;
+
+    public $attachButtonLabel;
+
+    public $attachedMessage;
+
+    public $manager;
+
+    public $owner;
+
+    public $related;
+
+    public function submit()
+    {
+        $this->owner->{$this->getRelationship()}()->attach($this->related);
+
+        $this->reset('related');
+
+        $this->emit('refreshRelationManagerList', $this->manager);
+
+        $this->dispatchBrowserEvent('close', "{$this->manager}RelationManagerAttachModal");
+        $this->dispatchBrowserEvent('notify', $this->attachedMessage);
+    }
+
+    public function getRelationship()
+    {
+        $manager = $this->manager;
+
+        return $manager::$relationship;
+    }
+
+    public function getInverseRelationship()
+    {
+        $manager = $this->manager;
+
+        if (property_exists($manager, 'inverseRelationship')) {
+            return $manager::$inverseRelationship;
+        }
+
+        return (string) Str::of(class_basename($this->owner))
+            ->lower()
+            ->plural()
+            ->camel();
+    }
+
+    public function getDisplayColumnName()
+    {
+        $manager = $this->manager;
+
+        return $manager::$displayColumnName;
+    }
+
+    public function mount()
+    {
+        $this->fillWithFormDefaults();
+    }
+
+    public function getForm()
+    {
+        $form = Form::make()
+            ->schema([
+                Select::make('related')
+                    ->label((string) Str::of($this->getRelationship())->singular()->title())
+                    ->placeholder('Start typing to search...')
```

Comment:
Is there an existing translation string for this?

Reply:
Imagine so, good spot.